# Design > Sample Size Comparison

<div class="alert alert-info">Calculate the required sample size for comparing two means or proportions</div>

The `sample_size_comp` function determines sample size requirements for A/B tests and other two-group comparisons. You can solve for any one of sample size, power, confidence level, effect size, or (for means) the standard deviation, by leaving that parameter as `None`.

In [1]:
import math

import pyrsm as rsm

In [2]:
## setup pyrsm for autoreload
%reload_ext autoreload
%autoreload 2
%aimport pyrsm

# Example: Comparing two proportions

You are running an A/B test for a website. The current conversion rate is 1.0% (p1 = 0.01). You want to detect if a new design reduces the conversion rate to 0.8% (p2 = 0.008). You want 90% power at a 95% confidence level, using a one-sided (less than) test.

How many visitors do you need in each group?

In [3]:
ssc = rsm.design.sample_size_comp(
    "proportion",
    p1=0.008,
    p2=0.01,
    conf=0.95,
    power=0.9,
    alternative="less",
)
ssc.summary()

Sample size calculation for comparison of proportions
Sample size 1    : 38,073
Sample size 2    : 38,073
Total sample size: 76,146
Proportion 1     : 0.008
Proportion 2     : 0.01
Effect size      : 0.021
Confidence level : 0.95
Power            : 0.9
Alternative      : less


You need approximately **38,073** visitors per group (about 76,146 total) to reliably detect a difference this small. Small effect sizes require large sample sizes.

# Solving for power

If you already know your sample sizes and want to find out how much power you have, leave `power` as `None`:

In [ ]:
ssc_power = rsm.design.sample_size_comp(
    "proportion",
    n1=38073,
    n2=38073,
    p1=0.008,
    p2=0.01,
    conf=0.95,
    alternative="less",
)
ssc_power.summary()

As expected, with 38,073 per group we get approximately 90% power.

# Example: Comparing two means

You want to test whether a new training program improves test scores by at least 5 points. The standard deviation of test scores is 10. You want 80% power with a two-sided test at 95% confidence.

In [ ]:
ssc_mean = rsm.design.sample_size_comp(
    "mean",
    delta=5,
    sd=10,
    conf=0.95,
    power=0.8,
    alternative="two-sided",
)
ssc_mean.summary()

# Power curve

The `plot()` method generates a power curve showing how power changes with sample size. The dashed red line marks the calculated sample size:

In [ ]:
ssc_mean.plot()

# Solving for confidence level

Given fixed sample sizes and desired power, you can also solve for the required confidence level by leaving `conf` as `None`:

In [ ]:
ssc_conf = rsm.design.sample_size_comp(
    "proportion",
    n1=38073,
    n2=38073,
    p1=0.008,
    p2=0.01,
    power=0.9,
    alternative="less",
)
ssc_conf.summary()

# Two-sided proportion test

For a two-sided test comparing conversion rates of 10% vs 15%:

In [ ]:
ssc_two = rsm.design.sample_size_comp(
    "proportion",
    p1=0.1,
    p2=0.15,
    conf=0.95,
    power=0.8,
    alternative="two-sided",
)
ssc_two.summary()

© Vincent Nijs (2026)